src/ui/common/src/components/workflows/operator/summaryList.tsx

```diff
@@ -0,0 +1,118 @@
+import { faChevronRight } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Link, List, ListItem } from '@mui/material';
+import Accordion from '@mui/material/Accordion';
+import AccordionDetails from '@mui/material/AccordionDetails';
+import AccordionSummary from '@mui/material/AccordionSummary';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import React, { useState } from 'react';
+import { Link as RouterLink } from 'react-router-dom';
+
+import { OperatorResultResponse } from '../../../handlers/responses/operator';
+import { getPathPrefix } from '../../../utils/getPathPrefix';
+import { OperatorType } from '../../../utils/operators';
+import { operatorTypeToIconMapping } from '../nodes/nodeTypes';
+
+type Props = {
+  title: string;
+  workflowId: string;
+  dagResultId: string;
+  operatorResults: OperatorResultResponse[];
+  initiallyExpanded: boolean;
+};
+
+const listStyle = {
+  width: '100%',
+  maxWidth: 360,
+  bgcolor: 'background.paper',
+};
+
+const SummaryList: React.FC<Props> = ({
+  title,
+  workflowId,
+  dagResultId,
+  operatorResults,
+  initiallyExpanded,
+}) => {
+  const [expanded, setExpanded] = useState(initiallyExpanded);
+  const items = operatorResults.map((opResult) => {
+    let link = `${getPathPrefix()}/workflow/${workflowId}/result/${dagResultId}/operator/${
+      opResult.id
+    }`;
+    const opType = opResult.spec?.type;
+    if (opType === OperatorType.SystemMetric || opType == OperatorType.Metric) {
+      link = `${getPathPrefix()}/workflow/${workflowId}/result/${dagResultId}/metric/${
+        opResult.id
+      }`;
+    }
+
+    if (opType === OperatorType.Check) {
+      link = `${getPathPrefix()}/workflow/${workflowId}/result/${dagResultId}/check/${
```

Comment:
@agiron123 I assume this is what we will have for check details page. It should automatically work once your changes get in

Reply:
@likawind route looks good here.